### PR TITLE
add external endpoint to retrieve available assistant configurations

### PR DESCRIPTION
**Pull request by fontanierh**
Most of the work is documentation

impl https://github.com/orgs/dust-tt/projects/3?pane=issue&itemId=40635114

### CODE REVIEW

File: docs/src/pages/agents.mdx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+## Agent Configuration Model
+
+The Agent Configuration model contains all the information and data related to an agent configuration.
+
+### Properties
```
Comment (fontanierh) [inline docs/src/pages/agents.mdx:5]: I omitted the action / generation config information as I don't think they are super relevant for end users at the moment (there's no POST).

File: docs/src/pages/agents.mdx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  <Property name="sId" type="string">
+    Unique identifier for the agent configuration.
+  </Property>
+  <Property name="scope" type="AgentConfigurationScope">
+    The scope of the agent configuration. It can be either `global` or
+    `workspace`.
+  </Property>
+  <Property name="status" type="AgentConfigurationStatus">
+    Status of the agent configuration. It can be one of the following: `active`,
+    `archived`, `disabled_by_admin`, `disabled_missing_datasource`,
```
Comment (fontanierh) [inline docs/src/pages/agents.mdx:17]: is that TMI  ?

File: docs/src/pages/agents.mdx (hunk 1 above)
Comment (fontanierh) [reply docs/src/pages/agents.mdx:5]: raises the question of whether we want to have separate output types for external / internal APIs. I lean towards no, but I think it makes sense to have parts of it that isn't documented

Comment (spolu) [reply docs/src/pages/agents.mdx:5]: Well for coherence we may want to have a rendering with lower_case and underscore... but we broke that rule for now.

By omitted you mean in the docs right? not in the API response?

File: front/pages/api/v1/w/[wId]/assistant/agent_configurations.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  if (keyRes.isErr()) {
+    return apiError(req, res, keyRes.error);
+  }
+
+  const { auth, keyWorkspaceId } = await Authenticator.fromKey(
+    keyRes.value,
+    req.query.wId as string
+  );
+
+  if (!auth.isBuilder() || keyWorkspaceId !== req.query.wId) {
```
Comment (spolu) [inline front/pages/api/v1/w/[wId]/assistant/agent_configurations.ts:25]: can be user

Comment (spolu) [reply front/pages/api/v1/w/[wId]/assistant/agent_configurations.ts:25]: even if not relevant/possible right now so I guess no-op

Comment (fontanierh) [reply front/pages/api/v1/w/[wId]/assistant/agent_configurations.ts:25]: API key == builder no ?

File: docs/src/pages/agents.mdx (hunk 2 above)
Comment (spolu) [reply docs/src/pages/agents.mdx:17]: That's fine

**Review by spolu — APPROVED** [review]

File: docs/src/pages/agents.mdx (hunk 1 above)
Comment (fontanierh) [reply docs/src/pages/agents.mdx:5]: yeah I meant docs

**Pull request merged**